src/shogun/features/CombinedDotFeatures.h

```diff
@@ -15,6 +15,8 @@
 #include <shogun/lib/DynamicObjectArray.h>
 #include <shogun/features/DotFeatures.h>
 
+#include <vector>
```

Comment:
please never include things like this is header files.... forward declare and include in .cpp file